source/dofs/dof_handler.cc

```diff
@@ -1156,6 +1156,9 @@ void DoFHandler<dim, spacedim>::distribute_mg_dofs (const FiniteElement<dim, spa
   (void)old_fe;
   Assert(old_fe == &fe, ExcMessage("You are required to use the same FE for level and active DoFs!") );
 
+  Assert((tria->get_mesh_smoothing() & Triangulation<dim, spacedim>::limit_level_difference_at_vertices)==true,
```

Comment:
The expression on the left is actually an `int`, so to compare it to `true` isn't quite right, strictly speaking. I think it might be clearer to compare for inequality it against `Triangulation::none`.